src/Sylius/Bundle/CoreBundle/PaymentRequest/Payum/Factory/PayumTokenFactory.php

```diff
@@ -33,7 +34,9 @@ public function createNew(PaymentRequestInterface $paymentRequest): TokenInterfa
         $paymentMethod = $paymentRequest->getMethod();
 
         $gatewayConfig = $paymentMethod->getGatewayConfig();
-        Assert::notNull($gatewayConfig, 'Gateway config cannot be null.');
+        if (null === $gatewayConfig) {
+            throw new NullGatewayConfigException();
```

Comment:
Isn't this not a case? Payment method cannot exists without gateway config unless something really wrong happened? Or new admin api allows this?

If not, then specialized exception feels overkill.